x-pack/plugins/observability/public/components/custom_threshold/components/rule_condition_chart/helpers.ts

```diff
@@ -0,0 +1,69 @@
+/*
+ * Copyright Elasticsearch B.V. and/or licensed to Elasticsearch B.V. under one
+ * or more contributor license agreements. Licensed under the Elastic License
+ * 2.0; you may not use this file except in compliance with the Elastic License
+ * 2.0.
+ */
+
+import {
+  Aggregators,
+  CustomThresholdExpressionMetric,
+} from '../../../../../common/custom_threshold_rule/types';
+
+export const getLensOperationFromRuleMetric = (metric: CustomThresholdExpressionMetric): string => {
+  const { aggType, field, filter } = metric;
+  let operation: string = aggType;
+  const operationArgs: string[] = [];
+  const aggFilter = JSON.stringify(filter || '').replace(/"|\\/g, '');
+
+  if (aggType === Aggregators.RATE) {
+    return `counter_rate(max(${field}), kql='${aggFilter}')`;
+  }
+
+  if (aggType === Aggregators.AVERAGE) operation = 'average';
+  if (aggType === Aggregators.CARDINALITY) operation = 'unique_count';
+  if (aggType === Aggregators.P95 || aggType === Aggregators.P99) operation = 'percentile';
+  if (aggType === Aggregators.COUNT) operation = 'count';
+
+  let sourceField = field;
+
+  if (aggType === Aggregators.COUNT) {
+    sourceField = '___records___';
+  }
+
+  operationArgs.push(sourceField || '');
+
+  if (aggType === Aggregators.P95) {
+    operationArgs.push('percentile=95');
+  }
+
+  if (aggType === Aggregators.P99) {
+    operationArgs.push('percentile=99');
+  }
+
+  if (aggFilter) operationArgs.push(`kql='${aggFilter}'`);
+
+  return operation + '(' + operationArgs.join(', ') + ')';
+};
+
+export const getBufferThreshold = (threshold?: number): string =>
+  (Math.ceil((threshold || 0) * 1.1 * 100) / 100).toFixed(2).toString();
+
+export const LensFieldFormat = {
+  NUMBER: 'number',
+  PERCENT: 'percent',
+  BITS: 'bits',
+} as const;
+
+export const lensFieldFormatter = (
+  metrics: CustomThresholdExpressionMetric[]
+): typeof LensFieldFormat[keyof typeof LensFieldFormat] => {
+  if (metrics.length < 1 || !metrics[0].field) return LensFieldFormat.NUMBER;
```

Comment:
@maryam-saeidi I agree. Even we can go further. Even if we have more than one metric, we check if ALL have the same field type and then apply the format; otherwise, there is no format.

Reply:
I agree, if all the fields have the same formatter, then we format, otherwise leave it raw.